control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -165,6 +166,17 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 	}
 
+	// Determine if the expected control plane operator image matches the active image in the control-plane-operator leader.
+	// If it doesn't need to no-op until the operator is recreated and at the matching version
+	if _, ok := hostedControlPlane.Annotations[hyperv1.ControlPlaneOperatorImageAnnotation]; !ok {
```

Comment:
Only need to access the index once:
```
expectedImage, ok := hostedControlPlane.Annotations[hyperv1.ControlPlaneOperatorImageAnnotation]
if !ok {
  return
}

if expectedImage != .ActiveImage {
```

Reply:
If we end up moving forward with current approach, we'll want this to be a unit tested function e.g `isCPOCompatibleWithHCP()`

Reply:
done